app/uk/gov/hmrc/gform/validation/ValidationUtil.scala

```diff
@@ -121,15 +121,18 @@ object ValidationUtil {
       case IsAddress(Address(_, _, _, Some(AuthCtx(AuthInfo.ItmpAddress)))) =>
         val itmpAddress = formModelVisibilityOptics.recalculationResult.evaluationContext.thirdPartyData.itmpRetrievals
           .flatMap(_.itmpAddress)
-        val atomMap: Map[String, String] = Map(
+        val atomMap1: Map[String, String] = Map(
           "street1"  -> itmpAddress.flatMap(_.line1).getOrElse(""),
           "street2"  -> itmpAddress.flatMap(_.line2).getOrElse(""),
           "street3"  -> itmpAddress.flatMap(_.line3).getOrElse(""),
           "street4"  -> itmpAddress.flatMap(address => address.line4.map(_ + " ") |+| address.line5).getOrElse(""),
           "postcode" -> itmpAddress.flatMap(_.postCode).getOrElse(""),
-          "country"  -> "",
-          "uk"       -> "true"
+          "uk"       -> itmpAddress.flatMap(_.countryName).filter(_.trim.nonEmpty).fold(true)(_ => false).toString
         )
+        val countryMap =
```

Comment:
This doesn't need to be in extra map. Let's put it this into `atomMap1` (and let's rename `atomMap1` to `atomMap`)